src/util/timemanager.cpp

```diff
@@ -412,6 +418,96 @@ void TimeManager::setDeltaTimeSteps(std::vector<double> deltaTimes) {
+        Key::Num3,
+        Key::Num4,
+        Key::Num5,
+        Key::Num6,
+        Key::Num7,
+        Key::Num8,
+        Key::Num9,
+        Key::Num0
+    };
+
+    // First, clear any previous keybinds
+    for (const KeyWithModifier& kb : _deltaTimeStepKeybindings) {
+        global::keybindingManager.removeKeyBinding(kb);
+    }
+    _deltaTimeStepKeybindings.clear();
+    _deltaTimeStepKeybindings.reserve(_deltaTimeSteps.size());
+
+    // Find positive delta time steps
+    std::vector<double> steps;
+    const int nStepsGuess = static_cast<int>(std::floor(_deltaTimeSteps.size() * 0.5f));
+    steps.reserve(nStepsGuess);
+    std::copy_if(
+        _deltaTimeSteps.begin(),
+        _deltaTimeSteps.end(),
+        std::back_inserter(steps),
+        [](double value) { return value >= 0.0; }
+    );
+
+    const int nKeys = static_cast<int>(Keys.size());
+    const int nSteps = static_cast<int>(steps.size());
+    const int maxIterations = (nSteps >= nKeys) ? nKeys : nSteps;
+
+    const char* guiPath = DeltaTimeStepsKeybindsGuiPath;
+
+    auto addDeltaTimeKeybind = [&guiPath, this](Key key, KeyModifier mod, double step) {
+        const std::string s = fmt::format("{:.0f}", step);
+        global::keybindingManager.bindKeyLocal(
+            key,
+            mod,
+            "openspace.time.interpolateDeltaTime(" + s + ")",
+            "Setting the simulation speed to " + s + " seconds per realtime second",
+            "Set Simulation Speed: " + s,
+            guiPath
+        );
+        _deltaTimeStepKeybindings.push_back(KeyWithModifier{ key, mod });
+    };
+
+    // For each key, add upp to three keybinds (no modifier, then SHIFT and then CTRL),
+    // plus inverted version of each time step one using the ALT modifier
+    for (int i = 0; i < maxIterations; ++i) {
+        const Key key = Keys[i];
+        addDeltaTimeKeybind(key, KeyModifier::NoModifier, steps[i]);
+        addDeltaTimeKeybind(key, KeyModifier::Alt, -steps[i]);
+
+        if (nSteps > nKeys) {
+            const int index = nKeys + i;
+            addDeltaTimeKeybind(key, KeyModifier::Shift, steps[index]);
+
+            KeyModifier mod = KeyModifier::Shift | KeyModifier::Alt;
+            addDeltaTimeKeybind(key, mod, -steps[index]);
+        }
+
+        if (nSteps > 2 * nKeys) {
+            const int index = 2 * nKeys + i;
+            addDeltaTimeKeybind(key, KeyModifier::Control, steps[index]);
+
+            KeyModifier mod = KeyModifier::Control | KeyModifier::Alt;
+            addDeltaTimeKeybind(key, mod, -steps[index]);
+        }
+    }
+
+    LINFO("Added keybindings for specified delta time steps.");
+    const int maxKeyBinds = 3 * nKeys;
+    if (nSteps > maxKeyBinds) {
+        LWARNING(fmt::format(
+            "Error settings delta time keys: Too many delta times, so not all could be "
+            "mapped to a key. Total: {} steps, which is {} more than the number of "
+            "available keybindings.",
+            nSteps,
```

Comment:
For the logging, etc, the "one parameter per line rule" is relaxed and you can put:
```
       ....
        nSteps, nSteps - maxKeybinds
```